gr-dtv/lib/atsc/atsc_plinfo.cc

```diff
@@ -0,0 +1,109 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2001,2006,2014 Free Software Foundation, Inc.
+ *
+ * This file is part of GNU Radio
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ */
+
+#include <gnuradio/dtv/atsc_plinfo.h>
+
+namespace gr {
+namespace dtv {
+
+plinfo::plinfo() : _flags(0), _segno(0){};
+plinfo::plinfo(uint16_t flags, int16_t segno) : _flags(flags), _segno(segno){};
+
+void plinfo::reset()
+{
+    _flags = 0;
+    _segno = 0;
+}
+
+void plinfo::from_tag_value(uint32_t tag_value)
+{
+    auto native_val = boost::endian::big_to_native(tag_value);
```

Comment:
Why not `htonl()`, to avoid dependency on boost?

Reply:
We already have the dependency on boost::endian via pmt, which is already cross platform.  htonl is very reasonable but require the inclusion of different headers depending on Linux/Windows/Mac, and hasn't yet been used in the codebase.